docs/src/understanding_mooncake/algorithmic_differentiation.md

```diff
@@ -178,16 +178,17 @@ Using the usual definition of the inner product between matrices,
 ```
 we can rearrange the inner product as follows:
 ```math
-\begin{align}
-    \langle \bar{Y}, D f [X] (\dot{X}) \rangle &= \langle \bar{Y}, \dot{X}^\top X + X^\top \dot{X} \rangle \nonumber \\
-        &= \textrm{tr} (\bar{Y}^\top \dot{X}^\top X) + \textrm{tr}(\bar{Y}^\top X^\top \dot{X}) \nonumber \\
-        &= \textrm{tr} ( [\bar{Y} X^\top]^\top \dot{X}) + \textrm{tr}( [X \bar{Y}]^\top \dot{X}) \nonumber \\
-        &= \langle \bar{Y} X^\top + X \bar{Y}, \dot{X} \rangle. \nonumber
-\end{align}
-```
-We can read off the adjoint operator from the first argument to the inner product:
+\begin{align*}
+\langle\bar{Y},Df[X](\dot{X})\rangle & =\langle\bar{Y},\dot{X}^{\top}X+X^{\top}\dot{X}\rangle\\
+ & =\textrm{tr}(\bar{Y}^{\top}\left(\dot{X}^{T}X+X^{T}\dot{X}\right))\\
+ & =\textrm{tr}(\dot{X}^{\top}X\bar{Y}^{\top})+\textrm{tr}(\bar{Y}^{\top}X^{\top}\dot{X})\\
+ & =\langle\dot{X},X\bar{Y}^{\top}\rangle+\langle\bar{Y}^{\top}X^{\top},\dot{X}\rangle\\
+ & =\langle X\bar{Y}^{\top}+X\bar{Y},\dot{X}\rangle.
+\end{align*}
+```
+The linearity of inner products and trace, and the [cyclic property of trace](https://en.wikipedia.org/wiki/Trace_(linear_algebra)#Cyclic_property) was used in the above. We can read off the adjoint operator from the first argument to the inner product:
 ```math
-D f [X]^\ast (\bar{Y}) = \bar{Y} X^\top + X \bar{Y}.
+D f [X]^\ast (\bar{Y}) = \bar{Y}^{\top} X + X \bar{Y}.
```

Comment:
I think this now has a different typo haha -- the multiplication order of `X` and `\bar{Y}` has been swapped, and I don't believe it should have been.
```suggestion
D f [X]^\ast (\bar{Y}) = X (\bar{Y}^{\top} + \bar{Y}).
```

Reply:
Thanks for catching the typos. I put the simplification in the end. Please see latest commit.